x-pack/legacy/plugins/searchprofiler/public/templates/index.html

```diff
@@ -89,14 +89,14 @@
       </div>
 
       <!-- Code editor -->
-      <div ui-ace="{ onLoad : aceLoaded, mode: 'json' }" ng-model="query" class="ace_editor"></div>
+      <div id="SearchProfilerInput" data-test-subj="searchProfilerEditor" class="ace_editor"></div>
```

Comment:
This is a bit of a guess, but I believe `ng-model="query"` forced Angular to trigger a digest cycle whenever the input changed. This allowed the "Profile" button to change disabled state immediately based on the presence or absence of input in the original code.

With these changes, the state of the "Profile" button won't update until another digest is triggered, e.g. by focusing and blurring the "Index" or "Type" fields. Notice in this screenshot how it's enabled despite there being no input.

![image](https://user-images.githubusercontent.com/1238659/61671164-2f996080-ac9b-11e9-8300-ba483e131fc3.png)

Reply:
Very well caught 😄 ! Yeah, that is almost certainly what is happening. Will fix.